backend/src/main/java/com/bakdata/conquery/models/index/MapInternToExternMapper.java

```diff
@@ -65,22 +76,37 @@ public class MapInternToExternMapper extends NamedImpl<InternToExternMapperId> i
 	//Manager only
 	@JsonIgnore
 	@Getter(onMethod_ = {@TestOnly})
-	private MapIndex int2ext = null;
+	private CompletableFuture<MapIndex> int2ext = null;
 
 
 	@Override
-	public synchronized void init() {
+	public synchronized MapInternToExternMapper init() {
+
+		if (mapIndex == null && config == null) {
+			log.trace("Injections were null. Skipping init, because class was deserialized by a test object mapper");
+			return this;
+		}
+
+		dataset = storage.getDataset();
 
 		final URI resolvedURI = FileUtil.getResolvedUri(config.getIndex().getBaseUrl(), csv);
 		log.trace("Resolved mapping reference csv url '{}': {}", this.getId(), resolvedURI);
 
-		int2ext = mapIndex.getIndex(new MapIndexKey(resolvedURI, internalColumn, externalTemplate));
+		MapIndexKey key = new MapIndexKey(resolvedURI, internalColumn, externalTemplate);
+
+		int2ext = CompletableFuture.supplyAsync(() -> mapIndex.getIndex(key)).whenComplete((m, e) -> {
+			if (e != null) {
+				log.warn("Unable to get index: {} (enable TRACE for exception)", key, (Exception) (log.isTraceEnabled() ? e : null));
```

Comment:
Hier auch bitte die Exception mitloggen. Meistens ist der Fehler fehlerhafte Konfiguration und die würden wir dann nicht sehen.

Reply:
Es könnte dann besser sein, wenn die die exception aus der vorherigen Konversation nicht geloggt wird, denn so haben wird viele Doppelungen in den logs

Reply:
@awildturtok ich habe jetzt eine custom exception eingeführt, damit das Fehlerhandling offensichtlicher ist, wenn man einen Index anfordert.